src/Compilers/CSharp/Portable/Symbols/Source/SourceMemberContainerSymbol.cs

```diff
@@ -3062,9 +3062,15 @@ private void AddSynthesizedRecordMembersIfNecessary(MembersAndInitializersBuilde
             var thisEquals = addThisEquals(equalityContract);
             addOtherEquals();
             addObjectEquals(thisEquals);
-            addGetHashCode(equalityContract);
+            var getHashCode = addGetHashCode(equalityContract);
             addEqualityOperators();
 
+            if ((thisEquals is not SynthesizedRecordEquals) ^ (getHashCode is not SynthesizedRecordGetHashCode))
```

Comment:
For regular classes, we only warn if equals was overridden but not hashcode. We don't warn if just GetHashCode is overridden. I believe that behavior is both fine and desirable, since Equals has implications on GetHashCode but GetHashCode does not have implications on equals. I think we should have the same behavior here.

Reply:
Related discussion: https://github.com/dotnet/roslyn/pull/47515#issuecomment-691279791

Reply:
Ah, I did miss that discussion, thanks. imo, we should have identical behavior. If we want to introduce a warning for GetHashCode without Equals, we should do so globally in a warning wave, not partly for just record types. Record classes have exactly the same equality/hashcode caveats as structs would today, so I don't think we're really gaining anything other than potentially making it harder for users to move between records and classes in a future C# version when we enable with as a pattern.

Reply:
That works for me. @jcouv what do you think?

Reply:
I agree. We should be consistent with `class` declarations here.

Reply:
@jaredpar The case here is a bit different than the warning for classes, see https://github.com/dotnet/roslyn/pull/47515#discussion_r490811870

Do we still need to have similar behavior, i.e. only warn if Equals is overridden but not GetHashCode?

Reply:
Yes, I would still expect to have similar behavior.

Reply:
> 
> 
> Yes, I would still expect to have similar behavior.

Updated to match the behavior.

Reply:
I'm good with that. Thanks